openassessment/xblock/team_mixin.py

```diff
@@ -0,0 +1,82 @@
+""" Mixin to encapsulate teams-related behavior """
+import logging
+
+from django.utils.functional import cached_property
+from django.core.exceptions import ObjectDoesNotExist
+from xblock.exceptions import NoSuchServiceError
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class TeamMixin(object):
+    """Team Mixin introducing all teams-related functionality."""
+
+    STAFF_OR_PREVIEW_INFO = {
+        'team_name': '< TEAM NAME >',
+        'team_usernames': ['< USERNAMES >'],
+        'team_url': '#TEAM_URL',
+    }
+    @cached_property
+    def teams_service(self):
+        try:
+            return self.runtime.service(self, 'teams')
+        except NoSuchServiceError:
+            logger.error(u'{}: Teams service unavailable'.format(self.location))
+
+    @cached_property
+    def team(self):
+        """
+        If the user associated with self.
+        Returns the CourseTeam model that represents the team or None if th user is not on a team
+
+        Raises:
+            - NoSuchServiceError if the teams service is unavailable
+            - ObjectDoesNotExist if the user associated with `anonymous_user_id`
+                                    can not be found
+        """
+        anonymous_user_id = self.get_anonymous_user_id_from_xmodule_runtime()
+        user = self.get_real_user(anonymous_user_id)
+        if not user:
+            logger.error(u'{}: User lookup for anonymous_user_id {} failed'.format(self.location, anonymous_user_id))
+            raise ObjectDoesNotExist()
+        team = self.teams_service.get_team(user, self.course_id)
+        return team
+
+    def has_team(self):
+        """
+        returns true if the student is on a team, false if the student is not on a team
+        or if an exception is raised while looking up team
+        """
+        try:
+            team = self.team
+        except Exception:  # pylint: disable=broad-except
+            return False
+        return bool(team)
+
+    def should_hide_team_ora(self):
+        """
+        Hide the ORA when the user is not on a team for a team ORA.
+        Ignore and return dummy data if we are course staff or in studio preview, since
+        we will want to see how the page will look to students
+        """
+        return (
+            (not self.has_team()) and not
```

Comment:
Nit: putting the second `not` on the next line would make it more clear that it applies to the disjunction on the next line.

Reply:
it's a pep8 violation to both have a newline before `and` and after `and`, so I just put everything on one line